vm/array.go

```diff
@@ -167,6 +167,59 @@ func builtinArrayInstanceMethods() []*BuiltinMethodObject {
 			},
 		},
 		{
+			// Passes each element of the collection to the given block. The method returns true if
+			// the block ever returns a value other than false or nil
+			//
+			// ```ruby
+			// a = [1, 2, 3]
+			//
+			// a.any? do |e|
+			//   e == 2
+			// end            # => true
+			// a.any? do |e|
+			//   e
+			// end            # => true
+			// a.any? do |e|
+			//   e == 5
+			// end            # => false
+			// a.any? do |e|
+			//   nil
+			// end            # => false
+			//
+			// a = []
+			//
+			// a.any? do |e|
+			//   true
+			// end            # => false
+			// ```
+			Name: "any?",
+			Fn: func(receiver Object) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *callFrame) Object {
+					arr := receiver.(*ArrayObject)
+
+					if blockFrame == nil {
+						return t.vm.initErrorObject(errors.InternalError, errors.CantYieldWithoutBlockFormat)
+					}
+
+					for _, obj := range arr.Elements {
+						result := t.builtinMethodYield(blockFrame, obj)
+
+						booleanResult, isResultBoolean := result.Target.(*BooleanObject)
+
+     				if isResultBoolean {
```

Comment:
Please fix these two line's indentation

Reply:
Fixed. Sorry for that - my editor hides differences between spaces and tabs, so the code looks normal to me when committing. I actually have a proposal about how to make git handle this automatically.

Reply:
OK, ignore the proposal. I will try to find, for this project, some way to automate the conversion of spaces to tabs.